### PR TITLE
pkp/pkp-lib#10336 Expose publication forms via API

### PR DESCRIPTION
**Pull request by taslangraham**
For https://github.com/pkp/pkp-lib/issues/10336

### CODE REVIEW

File: api/v1/submissions/PKPSubmissionController.php [hunk 1]
```diff
@@ -1706,4 +1754,236 @@ protected function getWriteDisabledErrors(string $schemaName, array $params): ar
...
+            ->values()
+            ->toArray();
+
+        return $config;
+    }
+
+    /***
+     * Get the Context for a given submission.
+     */
+    protected function getContext(PKPRequest $request, Submission $submission)
```
Comment (ewhanson) [inline api/v1/submissions/PKPSubmissionController.php:1934]: Return type is missing.

File: api/v1/submissions/PKPSubmissionController.php [hunk 2]
```diff
@@ -1706,4 +1754,236 @@ protected function getWriteDisabledErrors(string $schemaName, array $params): ar
...
+        $locales = $this->getPublicationFormLocales($data['context'], $data['submission']);
+        $titleAbstract = new TitleAbstractForm($data['publicationApiUrl'], $locales, $data['publication']);
+
+        return response()->json($titleAbstract->getConfig(), Response::HTTP_OK);
+    }
+
+    /**
+     * Utility method used to get the metadata locale information for a submission publications and context
+    */
+    protected function getPublicationFormLocales(Context $context, Submission $submission)
```
Comment (ewhanson) [inline api/v1/submissions/PKPSubmissionController.php:1890]: Return type is missing.

**Review by ewhanson — CHANGES_REQUESTED** [review]

File: api/v1/submissions/PKPSubmissionController.php (hunk 1 above)
Comment (taslangraham) [reply api/v1/submissions/PKPSubmissionController.php]: updated

File: api/v1/submissions/PKPSubmissionController.php (hunk 2 above)
Comment (taslangraham) [reply api/v1/submissions/PKPSubmissionController.php]: updated

**Review by ewhanson — APPROVED** [review]

**Pull request merged**